Cargo.toml

```diff
@@ -62,14 +62,17 @@ path = "sdk/core/azure_core_amqp"
 [workspace.dependencies.azure_core_test]
 # azure_core_test is not published and only ever a dev-dependency
 path = "sdk/core/azure_core_test"
+version = "0.1.0"
 
 [workspace.dependencies.azure_core_test_macros]
 # azure_core_test_macros is not published and only ever a dev-dependency
 path = "sdk/core/azure_core_test_macros"
+version = "0.1.0"
```

Comment:
Is this necessary for these two? They should never be published - and are authored as such - but not having a version helps doubly make sure they don't. If it simplifies things, though, no worries. I intentionally didn't add a `version` for that reason, though. They should only ever be `path` dependencies (or inherited, which equates to `path`).

Reply:
We should consult the `publish` for each crate, but if that's more work maybe we just exempt these two explicitly with a comment in the cargo script as to why. One less thing we have to worry about updating.

Reply:
We should exclude those.  I have it skipping updates to Cargo.tomls for packages with `publish = false`, but I should also exclude those packages from contributing versions for update

Reply:
I've pushed an update where we only pull package versions from tomls with `[package]` and not `publish = false`